engine/execution/computation/committer/committer.go

```diff
@@ -0,0 +1,71 @@
+package committer
+
+import (
+	"fmt"
+	"sync"
+
+	execState "github.com/onflow/flow-go/engine/execution/state"
+	"github.com/onflow/flow-go/fvm/state"
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+)
+
+type LedgerViewCommitter struct {
+	ldg    ledger.Ledger
+	tracer module.Tracer
+}
+
+func NewLedgerViewCommitter(ldg ledger.Ledger, tracer module.Tracer) *LedgerViewCommitter {
+	return &LedgerViewCommitter{ldg: ldg, tracer: tracer}
+}
+
+func (s *LedgerViewCommitter) CommitView(view state.View, baseState flow.StateCommitment) (newCommit flow.StateCommitment, proof []byte, err error) {
+	var err1, err2 error
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		proof, err2 = s.collectProofs(view, baseState)
+		wg.Done()
+	}()
+
+	newCommit, err1 = s.commitView(view, baseState)
+	wg.Wait()
+
+	if err1 != nil {
```

Comment:
Can use a multierror to capture and return both errors here

Reply:
fixed at fd893bd